phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java

```diff
@@ -78,8 +80,7 @@ public TableSnapshotResultIterator(Configuration configuration, Scan scan, ScanM
     this.scan = scan;
     this.scanMetricsHolder = scanMetricsHolder;
     this.scanIterator = UNINITIALIZED_SCANNER;
-    this.restoreDir = new Path(configuration.get(PhoenixConfigurationUtil.RESTORE_DIR_KEY),
-        UUID.randomUUID().toString());
+    this.restoreDir = new Path(configuration.get(PhoenixConfigurationUtil.RESTORE_DIR_KEY));
```

Comment:
Will it change any semantics  if we are not adding uuid to the restoreDir path when isMRSnapshotManagedExternally is set to false ?

Reply:
@shahrs87 I don't think so. I believe the original flow was faulty in itself and for every scan, we were creating a subdirectory, not instead of that, we have the same subdirectory for restore and every scan restore happens there and clean up also happens per scan. So there should not be any issue. We want to get rid of this structure because when we do an external restore or external cleanup, we need to provide exact restore directory for reading it.

Reply:
@sakshamgangwar - Remember that scans are taking place in parallel as multiple mappers may be running at the same time. Until we fix the automation to only create once in a future JIRA, don't we need to keep the restored files in separate directories so tasks won't step on each other?

Reply:
@gjacoby126I missed that part, thanks for catching that. I will guard the directories with the same config as well.

Reply:
@gjacoby126 Can you review again? I have kept old behavior as it is, as well as in the assertion for restore directory I am asserting whether or not snapshot directory exists (in case of externally cleaned up) or not (cleaned up per scan).